Optionally install PyObjC if on macOS test env

verbosus opened the pull request:
Trying my luck with GHA conditional syntax (is there a way to see what a workflow change would do without pushing to github?) in order to fix #2108. The updated workflow should `pip install pyobjc` only on macOS.

verbosus commented:
OK, so if the PR gets approved then the GHA runner will actually execute and we’ll get to see if the changes where any good? 🤪

anthrotype commented:
the GH action workflow has not run because it says "invalid workflow file".. There seems to be a syntax error somewhere

https://github.com/fonttools/fonttools/actions/runs/374300397

anthrotype commented:
whatever the syntax error in the yaml config, I think the best way to pip install some test dependency is to update the `requirements.txt` file

https://github.com/fonttools/fonttools/blob/master/requirements.txt

and add the following conditional requirement:

`pyobjc; sys_platform == 'darwin'"`

chrissimpkins commented:
> the GH action workflow has not run because it says "invalid workflow file".. There seems to be a syntax error somewhere
> 
> 
> 
> https://github.com/fonttools/fonttools/actions/runs/374300397

It looks like you need single quotes on string literals in expressions.  🤷‍♂ 

https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#literals

verbosus commented:
OK, I reverted the change in the GHA workflow and updated requirements.txt as per Cosimo’s request (I pinned the version to the latest one which works here.)
